> > > > You could also consider that sys/spawn.h is included in legacy support and might fix the tests. But if this is the easier fix, I am not picky.
> > > 
> > > 
> > > That would be a much better fix.
> > 
> > 
> > I'm certain I tired legacy_support and it did not provide spawn.h. How are you adding legacy_support @ForestExpertise ? Maybe I didn't do it in the same way or something
> 
> The trick is that legacy-support does not contain spawn.h, it contains sys/spawn.h So everywhere spawn.h is included, you have to change the include to sys/spawn.h instead. And include legacy support in the portfile, of course. https://github.com/macports/macports-legacy-support/blob/master/include/sys/spawn.h Not sure why legacy support did it this way, and it does create a hassle, but works for most things. I recall there is a function they didn't implement in spawn.h, so some of your workarounds may still be needed. But it is worth trying if you have time

Won't work for this unfortunately. Perhaps this is why they don't have it as a drop in: 
```bcommon-1.7.0 -I../libxkbcommon-1.7.0/src -I../libxkbcommon-1.7.0/include -I/opt/local/include/LegacySupport -I/opt/local/include -fdiagnostics-color=always -Wall -Winvalid-pch -Wextra -std=c11 -O0 -g -fno-strict-aliasing -Wno-unused-parameter -Wno-missing-field-initializers -Wpointer-arith -Wmissing-declarations -Wformat=2 -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Wbad-function-cast -Wshadow -Wlogical-op -Wdate-time -Wwrite-strings -pipe -Os -arch i386 -MD -MQ libxkbcommon-x11-internal.a.p/test_xvfb-wrapper.c.o -MF libxkbcommon-x11-internal.a.p/test_xvfb-wrapper.c.o.d -o libxkbcommon-x11-internal.a.p/test_xvfb-wrapper.c.o -c ../libxkbcommon-1.7.0/test/xvfb-wrapper.c
FAILED: libxkbcommon-x11-internal.a.p/test_xvfb-wrapper.c.o 
/opt/local/bin/gcc-mp-16 -Ilibxkbcommon-x11-internal.a.p -I. -I../libxkbcommon-1.7.0 -I../libxkbcommon-1.7.0/src -I../libxkbcommon-1.7.0/include -I/opt/local/include/LegacySupport -I/opt/local/include -fdiagnostics-color=always -Wall -Winvalid-pch -Wextra -std=c11 -O0 -g -fno-strict-aliasing -Wno-unused-parameter -Wno-missing-field-initializers -Wpointer-arith -Wmissing-declarations -Wformat=2 -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Wbad-function-cast -Wshadow -Wlogical-op -Wdate-time -Wwrite-strings -pipe -Os -arch i386 -MD -MQ libxkbcommon-x11-internal.a.p/test_xvfb-wrapper.c.o -MF libxkbcommon-x11-internal.a.p/test_xvfb-wrapper.c.o.d -o libxkbcommon-x11-internal.a.p/test_xvfb-wrapper.c.o -c ../libxkbcommon-1.7.0/test/xvfb-wrapper.c
../libxkbcommon-1.7.0/test/xvfb-wrapper.c: In function 'xvfb_wrapper':
../libxkbcommon-1.7.0/test/xvfb-wrapper.c:92:11: error: implicit declaration of function 'posix_spawnp'; did you mean 'posix_openpt'? [-Wimplicit-function-declaration]
   92 |     ret = posix_spawnp(&xvfb_pid, "Xvfb", NULL, NULL, xvfb_argv, envp);
      |           ^~~~~~~~~~~~
      |           posix_openpt
../libxkbcommon-1.7.0/test/xvfb-wrapper.c:92:11: warning: nested extern declaration of 'posix_spawnp' [-Wnested-externs]
../libxkbcommon-1.7.0/test/xvfb-wrapper.c: In function 'x11_tests_run':
../libxkbcommon-1.7.0/test/xvfb-wrapper.c:148:12: warning: variable 'count' set but not used [-Wunused-but-set-variable=]
  148 |     size_t count = 1; /* For NULL-terminated entry */
      |            ^~~~~
[87/205] /opt/local/bin/gcc-mp-16 -Ilibxkbcommon-x11-internal.a.p -I. -I../libxkbcommon-1.7.0 -I../libxkbcommon-1.7.0/src -I../libxkbcommon-1.7.0/include -I/opt/local/include/LegacySupport -I/opt/local/include -fdiagnostics-color=always -Wall -Winvalid-pch -Wextra -std=c11 -O0 -g -fno-strict-aliasing -Wno-unused-parameter -Wno-missing-field-initializers -Wpointer-arith -Wmissing-declarations -Wfor```

--- a/devel/libxkbcommon/Portfile
+++ b/devel/libxkbcommon/Portfile
@@ -65,6 +65,14 @@ subport ${name}-x11 {
                     port:${name} \
                     port:xorg-libxcb
 
+    # The tests enabled by -Denable-x11=true require <spawn.h>, which tiger
+    # does not have.
+    platform darwin 8 {
+        patchfiles-append \
+                test_x11comp.c_tiger.diff \
+                test_xvfb-wrapper.c_tiger.diff
+    }
+
     configure.args-replace \
                     -Denable-x11=false -Denable-x11=true
 

--- a/devel/libxkbcommon/files/test_x11comp.c_tiger.diff
+++ b/devel/libxkbcommon/files/test_x11comp.c_tiger.diff
@@ -1,0 +1,36 @@
+--- test/x11comp.c.orig	2026-08-17 15:58:17.000000000 -0600
++++ test/x11comp.c	2026-08-17 16:28:16.000000000 -0600
+@@ -24,7 +24,14 @@
+ #include "config.h"
+ 
+ #include <stdio.h>
+-#include <spawn.h>
++#if defined(__APPLE__)
++    #include "AvailabilityMacros.h"
++    #if MAC_OS_X_VERSION_MIN_REQUIRED > 1040
++        #include <spawn.h>
++    #endif
++#else
++    #include <spawn.h>
++#endif
+ #include <unistd.h>
+ #include <assert.h>
+ #include <signal.h>
+@@ -37,6 +44,9 @@
+ 
+ X11_TEST(test_basic)
+ {
++#if defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED < 1050
++    return 0;
++#else
+     struct xkb_keymap *keymap;
+     xcb_connection_t *conn;
+     int32_t device_id;
+@@ -114,6 +124,7 @@
+     xcb_disconnect(conn);
+ err_conn:
+     return ret;
++#endif
+ }
+ 
+ int main(void) {

--- a/devel/libxkbcommon/files/test_xvfb-wrapper.c_tiger.diff
+++ b/devel/libxkbcommon/files/test_xvfb-wrapper.c_tiger.diff
@@ -1,0 +1,36 @@
+--- test/xvfb-wrapper.c.orig	2026-08-17 16:14:33.000000000 -0600
++++ test/xvfb-wrapper.c	2026-08-17 16:29:02.000000000 -0600
+@@ -25,7 +25,14 @@
+ #include "config.h"
+ 
+ #include <stdio.h>
+-#include <spawn.h>
++#if defined(__APPLE__)
++    #include "AvailabilityMacros.h"
++    #if MAC_OS_X_VERSION_MIN_REQUIRED > 1040
++        #include <spawn.h>
++    #endif
++#else
++    #include <spawn.h>
++#endif
+ #include <assert.h>
+ #include <signal.h>
+ #include <sys/types.h>
+@@ -46,6 +53,9 @@
+ int
+ xvfb_wrapper(int (*test_func)(char* display))
+ {
++#if defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED < 1050
++    return 0;
++#else
+     int ret = 0;
+     FILE * display_fd;
+     char display_fd_string[32];
+@@ -134,6 +144,7 @@
+     fclose(display_fd);
+ err_display_fd:
+     return ret;
++#endif
+ }
+ 
+ /* All X11_TEST functions are in the test_func_sec ELF section.

